src/tmpfiles/tmpfiles.c

```diff
@@ -2896,6 +2896,26 @@ static int parse_age_by_from_arg(const char *age_by_str, Item *item) {
         return 0;
 }
 
+static bool is_duplicated_item(ItemArray *existing, Item *i) {
+
+        assert(existing);
+        assert(i);
+
+        for (size_t n = 0; n < existing->n_items; n++) {
+                Item *e = existing->items + n;
+
+                if (item_compatible(e, i))
+                        continue;
+
+                /* Only multiple 'w+' lines for the same path are allowed. */
+                if (e->type != WRITE_FILE || !e->append_or_force ||
+                    i->type != WRITE_FILE || !i->append_or_force)
+                    return true;
```

Comment:
The condition must be inverted.
```
if (e->type == WRITE_FILE && e->append_or_force &&
    i->type == WRITE_FILE && i->append_or_force)
        continue;

return true;
```

BTW, why not to add the condition into `item_compatible()`?

Reply:
Because It looked more readable but it's subjective so I've no strong argument.